compiler/sem/sigmatch.nim

```diff
@@ -2796,6 +2798,10 @@ proc matchesAux(c: PContext, n, nOrig: PNode, m: var TCandidate, marker: var Int
         m.error.firstMismatch.kind = kVarNeeded
         noMatch()
 
+  template acceptsTyped(typ: PType): bool =
+    typ != nil and (typ.kind == tyTyped or
+                    typ.kind == tyVarargs and typ[0].kind == tyTyped)
```

Comment:
```suggestion
    m.calleeSym != nil and m.calleeSym.kind in {skMacro, skTemplate} and
      typ != nil and (typ.kind == tyTyped or
                      typ.kind == tyVarargs and typ[0].kind == tyTyped)
```
This is not very pretty, but it does prevent `varargs[typed]` used outside of templates/macros (`echo`, for example) from treating an error as acceptable.

At the moment, the error would be leaked, and only later wrapped via `foldInAst`.

Reply:
Thank you, fixed this locally.

Reply:
`echo`'s definition is a bit abusive, we're treating `typed` almost like a `dynamic` keyword, where it's not just like an `any`, which binds to a single type, but each element is bound separately.

Mostly just noting an interesting case for the future.